rustls/src/verify.rs

```diff
@@ -642,6 +672,8 @@ fn pki_error(error: webpki::Error) -> Error {
         CertExpired | InvalidCertValidity => CertificateError::Expired.into(),
         UnknownIssuer => CertificateError::UnknownIssuer.into(),
         CertNotValidForName => CertificateError::NotValidForName.into(),
+        CertRevoked => CertificateError::Revoked.into(),
+        IssuerNotCrlSigner => CrlError::InvalidIssuer.into(),
 
         InvalidSignatureForPublicKey
         | UnsupportedSignatureAlgorithm
```

Comment:
Notably we can emit these errors when consulting a CRL for revocation information for a cert that has an issuer matching the CRL's issuer.

This is a bit suboptimal. For example, it might be confusing to have a `CertificateError::InvalidSignatureForPublicKey` be produced if it was the **CRL signature** that was invalid.

Perhaps that means on the webpki side we should add `InvalidCrlSignatureForPublicKey` and `UnsupportedCrlSignatureAlgorithm` errors that could be caught here and turned into `CrlError::xxxx` variants? WDYT?

Reply:
Done.